conda/activate.py

```diff
@@ -929,46 +907,40 @@ def _update_prompt(self, set_vars, conda_prompt_modifier):
             }
         )
 
-    def _hook_preamble(self):
-        result = ""
+    def _hook_preamble(self) -> str:
+        result = []
         for key, value in context.conda_exe_vars_dict.items():
             if value is None:
                 # Using `unset_var_tmpl` would cause issues for people running
                 # with shell flag -u set (error on unset).
-                # result += join(self.unset_var_tmpl % key) + '\n'
```

Comment:
That line is dropped, any idea why?

Reply:
What do you mean? That line is a comment line, the next line is converted to append and the `\n` handling moved to the end of the function, which is more generic.